pumps/resurface_test.go

```diff
@@ -0,0 +1,333 @@
+			RawRequest:  rawReq,
+			RawResponse: rawResp,
+			TimeStamp:   time.Now(),
+		},
+	}
+
+	err := pmp.WriteData(context.TODO(), recs)
+	if err != nil {
+		t.Fatal(pmp.GetName()+"couldn't write records with err:", err)
+	}
+
+	queue := pmp.logger.Queue()
+	assert.Equal(t, len(recs), len(queue))
+
+	for i, message := range queue {
+		if i < 3 {
+			assert.Contains(t, message, "custom_field:tyk-API-ID\",\""+recs[i].(analytics.AnalyticsRecord).APIID)
+			assert.Contains(t, message, "custom_field:tyk-Org-ID\",\""+recs[i].(analytics.AnalyticsRecord).OrgID)
+			if i == 2 {
+				assert.Contains(t, message, "custom_field:tyk-API-Key\",\""+recs[i].(analytics.AnalyticsRecord).APIKey)
+				assert.Contains(t, message, "custom_field:tyk-API-Name\",\""+recs[i].(analytics.AnalyticsRecord).APIName)
+				assert.Contains(t, message, "custom_field:tyk-API-Version\",\""+recs[i].(analytics.AnalyticsRecord).APIVersion)
+				assert.Contains(t, message, "custom_field:tyk-Oauth-ID\",\""+recs[i].(analytics.AnalyticsRecord).OauthID)
+			}
+		} else {
+			assert.NotContains(t, message, "custom_field:tyk")
+		}
+	}
+}
+
+func TestResurfaceWriteChunkedResponse(t *testing.T) {
+	pmp, _ := SetUp(t, "", make([]string, 0), "include debug")
+
+	recs := []interface{}{
+		analytics.AnalyticsRecord{
+			Host:        "test-three",
+			Method:      "GET",
+			RawRequest:  rawReq,
+			RawResponse: rawRespOneChunk,
+			TimeStamp:   time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			APIID:       "api-id-x",
+			OrgID:       "api-org-y",
+			Host:        "test-4",
+			Method:      "GET",
+			RawRequest:  rawReq,
+			RawResponse: rawRespChunks,
+			TimeStamp:   time.Now(),
+		},
+		analytics.AnalyticsRecord{
+			APIID:       "",
+			OrgID:       "",
+			Host:        "test.five",
+			Method:      "GET",
+			RawRequest:  rawReq,
+			RawResponse: rawRespChunksTrailer,
+			TimeStamp:   time.Now(),
+		},
+	}
+
+	err := pmp.WriteData(context.TODO(), recs)
+	if err != nil {
+		t.Fatal(pmp.GetName()+"couldn't write records with err:", err)
+	}
+
+	queue := pmp.logger.Queue()
+	assert.Equal(t, len(recs), len(queue))
+
+	for i, message := range queue {
+		assert.Contains(t, message, "request_url\",\"http://localhost:8080/get")
+		assert.NotContains(t, message, "response_header:content-length")
+		assert.Contains(t, message, "[\"response_header:transfer-encoding\",\"chunked\"]")
+		if i != 4 {
+			assert.Regexp(t, `\[\"response_body\",\".*\\r\\n0(?:\\r\\n)+\"\]`, message)
+		} else {
+			assert.Regexp(t, `\[\"response_body\",\".*\\r\\n0\\r\\nExpires:.*\"\]`, message)
+		}
+	}
+}
+
+func TestResurfaceSkipWrite(t *testing.T) {
+	pmp, _ := SetUp(t, "", make([]string, 0), "include debug")
+
+	recs := []interface{}{
+		analytics.AnalyticsRecord{
+			APIID:       "an-api-id",
+			OrgID:       "an-api-org",
+			Host:        "test6",
+			Method:      "POST",
+			RawRequest:  "",
+			RawResponse: "",
+			TimeStamp:   time.Now(),
+		},
+	}
+
+	err := pmp.WriteData(context.TODO(), recs)
+	if err != nil {
```

Comment:
We use `assert.NotNil` here

Reply:
Thank you, I just modified it accordingly.